Center info page

thebagmaster opened the pull request:
The design file was updated to show the cover for the onboarding prompt was changed to be centered. This required some cleanup in both in InformationPage as well as OnboardingCover. This small change should help with future usages of InformationPage.

The changes include:
- ~Removing states from InformationPage in favor of local bool~ Just some edits to the changes for the states
- Adding an encompassing ColumnLayout as well as a ~filler item~ < Bad practice
- Moving control for centering into OnboardingCover

`edit: see comments below for working screenshots.`

![ss](https://github.com/bitcoin-core/gui-qml/assets/7547321/c1fce122-c16d-4121-acf1-ad53c2df7fc1)

[![Windows](https://img.shields.io/badge/OS-Windows-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/win64/unsecure_win_gui.zip?branch=pull/345)
[![Intel macOS](https://img.shields.io/badge/OS-Intel%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos/unsecure_mac_gui.zip?branch=pull/345)
[![Apple Silicon macOS](https://img.shields.io/badge/OS-Apple%20Silicon%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos_arm64/unsecure_mac_arm64_gui.zip?branch=pull/345)
[![ARM64 Android](https://img.shields.io/badge/OS-Android-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/unsecure_android_apk.zip?branch=pull/345)
[![ARM32 Android](https://img.shields.io/badge/OS-Android%2032bit-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android32/unsecure_android_32bit_apk.zip?branch=pull/345)

<!--
*** Please remove the following help text before submitting: ***

Pull requests without a rationale and clear improvement may be closed
immediately.

GUI-related pull requests should be opened against
https://github.com/bitcoin-core/gui
first. See CONTRIBUTING.md
-->

<!--
Please provide clear motivation for your patch and explain how it improves
Bitcoin Core user experience or Bitcoin Core developer experience
significantly:

* Any test improvements or new tests that improve coverage are always welcome.
* All other changes should have accompanying unit tests (see `src/test/`) or
  functional tests (see `test/`). Contributors should note which tests cover
  modified code. If no tests exist for a region of modified code, new tests
  should accompany the change.
* Bug fixes are most welcome when they come with steps to reproduce or an
  explanation of the potential issue as well as reasoning for the way the bug
  was fixed.
* Features are welcome, but might be rejected due to design or scope issues.
  If a feature is based on a lot of dependencies, contributors should first
  consider building the system outside of Bitcoin Core, if possible.
* Refactoring changes are only accepted if they are required for a feature or
  bug fix or otherwise improve developer experience significantly. For example,
  most "code style" refactoring changes require a thorough explanation why they
  are useful, what downsides they have and why they *significantly* improve
  developer experience or avoid serious programming bugs. Note that code style
  is often a subjective matter. Unless they are explicitly mentioned to be
  preferred in the [developer notes](/doc/developer-notes.md), stylistic code
  changes are usually rejected.
-->

<!--
Bitcoin Core has a thorough review process and even the most trivial change
needs to pass a lot of eyes and requires non-zero or even substantial time
effort to review. There is a huge lack of active reviewers on the project, so
patches often sit for a long time.
-->

<!--
Links for Windows and macOS build artifacts. Replace <PR> with the assigned pull request number.

[![Windows](https://img.shields.io/badge/OS-Windows-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/win64/unsecure_win_gui.zip?branch=pull/<PR>)
[![Intel macOS](https://img.shields.io/badge/OS-Intel%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos/unsecure_mac_gui.zip?branch=pull/<PR>)
[![Apple Silicon macOS](https://img.shields.io/badge/OS-Apple%20Silicon%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos_arm64/unsecure_mac_arm64_gui.zip?branch=pull/<PR>)
[![ARM64 Android](https://img.shields.io/badge/OS-Android-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/unsecure_android_apk.zip?branch=pull/<PR>)
[![ARM32 Android](https://img.shields.io/badge/OS-Android%2032bit-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android32/unsecure_android_32bit_apk.zip?branch=pull/<PR>)
-->

jarolrod commented on src/qml/pages/onboarding/OnboardingCover.qml:
```diff
@@ -48,6 +51,7 @@ Page {
             descriptionSize: 24
             subtext: qsTr("100% open-source & open-design")
             buttonText: qsTr("Start")
+            centerContent: AppMode.state=="MOBILE"
```
this can be AppMode.isDesktop instead

jarolrod reviewed (commented):
from linter: 

```
This diff appears to have added new lines with trailing whitespace.
The following changes were suspected:

diff --git a/src/qml/controls/InformationPage.qml b/src/qml/controls/InformationPage.qml
@@ -52 +53,6 @@ Page {
+        
@@ -93,0 +97,5 @@ Page {
+            Item { 
+              Layout.fillHeight: !root.centerContent 
```

johnny9 commented on src/qml/controls/InformationPage.qml:
```diff
@@ -102,46 +110,9 @@ Page {
-    state: AppMode.state
-
-    states: [
-        State {
```
The states are required to handle the anchoring differences between mobile and desktop. anchors changes should be handled through States/AnchorChanges. https://doc.qt.io/qt-6/qtquick-positioning-anchors.html#changing-anchors)](https://doc.qt.io/qt-6/qtquick-positioning-anchors.html#changing-anchors

johnny9 commented on src/qml/controls/InformationPage.qml:
```diff
@@ -83,14 +87,18 @@ Page {
                 Layout.maximumWidth: detailMaximumWidth
                 sourceComponent: root.detailItem
             }
+            Item { 
```
Use AnchorChanges instead of adding a useless Item.

thebagmaster commented:
Rebased
Added the states back in
Had to respect button position for desktop should behave like design doc again
![SS1](https://github.com/bitcoin-core/gui-qml/assets/7547321/cc2ef64f-e4fe-432b-85da-556a3c432be6)
![SS2](https://github.com/bitcoin-core/gui-qml/assets/7547321/f789319f-837b-4295-a23e-2c0d87af69b4)

jarolrod commented:
Closing for now; this needs to be rebased over main, but we'll be taking a different approach as to the view components and how we handle desktop vs mobile
